ui/app/src/components/FormsEngine/controls/Textarea.tsx

```diff
@@ -14,29 +14,56 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import OutlinedInput from '@mui/material/OutlinedInput';
-import React, { useId } from 'react';
+import OutlinedInput, { OutlinedInputProps } from '@mui/material/OutlinedInput';
+import React, { useEffect, useId, useMemo } from 'react';
 import { FormsEngineField } from '../components/FormsEngineField';
 import { ControlProps } from '../types';
+import { escapeXml, unescapeXml } from '../../../utils/xml';
+import { nnou, nou } from '../../../utils/object';
 
 export interface TextareaProps extends ControlProps {
 	value: string;
 }
 
 export function Textarea(props: TextareaProps) {
-	const { field, value, setValue, readonly, autoFocus } = props;
+	const { field, value: valueProp, setValue, readonly: formReadonly, autoFocus } = props;
 	const htmlId = useId();
-	const maxLength = field.validations.maxLength?.value;
+
+	// region field properties/validations
+	const maxLength = field.validations?.maxLength?.value;
+	const readonly = formReadonly || (field.properties?.readonly?.value as boolean);
+	const escapeContent = (field.properties?.escapeContent?.value as boolean) ?? false;
+	const rows = (field.properties?.rows?.value as number) ?? 1;
+	const defaultValue = field.defaultValue as string;
+	// endregion
+
+	const value = useMemo(() => {
+		const hasValue = nnou(valueProp); // allows '' as a valid value
+		return hasValue ? (escapeContent ? unescapeXml(valueProp) : valueProp) : (defaultValue ?? '');
+	}, [valueProp, escapeContent, defaultValue]);
+
+	useEffect(() => {
+		// If there's a default value and no value has been set yet, set it as the value.
+		if (nou(valueProp) && defaultValue) {
+			setValue(defaultValue);
+		}
+	}, [defaultValue, setValue, valueProp]);
```

Comment:
Could we do this in batch at the form initialisation? Aren't we doing it already?